src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -145,14 +145,12 @@ private ReadFilterLibrary(){ /*no instance*/ }
         @Override public boolean test(final GATKRead read) {
             return read.isProperlyPaired();}}
 
-    //TODO: should this reject supplementary alignments ? If not it should be
-    //removed since its redundant with NotSecondaryAlignmentReadFilter
-    //https://github.com/broadinstitute/gatk/issues/2165
+    //* this now tests for unmapped or neither secondary nor supplementary as its name suggests */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class PrimaryAlignmentReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
-            return ! read.isSecondaryAlignment();}}
+            return read.isUnmapped() || !(read.isSecondaryAlignment()||read.isSupplementaryAlignment());}}
```

Comment:
I was imitating what I saw in other filters that were part of Wellformed.  Cf., ValidAlignmentStartReadFilter, ValidAlignmentEndReadFilter, ReadUtils.alignmentAgreesWithHeader, ReadLengthEqualsCigarLengthReadFilter.
Are you sure you don't want me to accept unmapped reads?
It looks to me as if most every filter that tests alignment criteria accepts them.